src/big/big_decimal.cr

```diff
@@ -235,6 +255,67 @@ struct BigDecimal < Number
     end
   end
 
+  # Round to the nearest integer (by default), returning the result as a `BigDecimal`.
+  #
+  # ```
+  # BigDecimal.new("3.14159").round # => 3_big_d
+  # BigDecimal.new("8.7").round     # => 9_big_d
+  # BigDecimal.new("-9.9").round    # => -10_big_d
+  # ```
+  #
+  # If *digits* is specified and positive, the fractional part of the result
+  # has no more than that many digits.
+  #
+  # The value of the optional *mode* argument can be used to determine how
+  # rounding is performed.
+  def round(digits = 0, mode : RoundingMode = :half_up) : BigDecimal
+    return self.class.new(value, @scale) if @scale <= digits
+
+    n_scale = @scale - digits
+    n_digits = value.digits
+
+    # Workaround for "-3" in digits array
+    if n_digits.first?.try &.< 0
+      negative = true
+      n_digits.shift
+    end
+
+    msd = 0
+    n_scale.times do
+      msd = n_digits.pop? || 0
+    end
+    n_digits << 0 if n_digits.empty?
+    str_value = String.build do |str|
+      n_digits.each &.to_s(str)
+    end
+
+    n_value = BigInt.new(str_value)
+    return self.class.new(n_value, digits) if msd.zero?
+
+    round_up = case mode
+               when .up?        then true
+               when .down?      then false
+               when .half_up?   then msd >= 5
+               when .half_down? then msd > 5
+               when .half_even? then msd == 5 ? !n_digits.last.even? : msd > 5
+               when .ceil?      then !negative
+               when .floor?     then negative
+               else
+                 raise "unreachable"
+               end
+
+    bd = n_value
+    bd += 1 if round_up
+    bd = self.class.new(bd, digits)
+    bd *= -1 if negative
+    bd
+  end
+
+  # :nodoc:
+  def round(digits, mode : Symbol) : BigDecimal
```

Comment:
This definition is required due to enum autocasting bug. Otherwise user is left with `BigDecimal::RoundingMode::HALF_UP` - vs - `:half_up`